src/main/c/zlib/zlib.c

```diff
@@ -600,11 +600,11 @@ static void
 zstream_init(struct zstream *z, const struct zstream_funcs *func)
 {
     z->flags = 0;
-    z->buf = Qnil;
+    z->buf = rb_tr_handle_for_managed_leaking(Qnil);
     z->buf_filled = 0;
-    z->input = Qnil;
-    z->stream.zalloc = zlib_mem_alloc;
-    z->stream.zfree = zlib_mem_free;
+    z->input = rb_tr_handle_for_managed_leaking(Qnil);
```

Comment:
Can't we share handles for constants like `Qnil`?

Reply:
Good idea! Should be handled centrally in `rb_tr_handle_for_managed_leaking` though. We need a general re-think about what we're going to do with all these handles.

Reply:
We must be careful to not release such handles re-used in multiple places, maybe by handling them specially in `rb_tr_release_handle()`.